orsopy/fileio/base.py

```diff
@@ -82,7 +84,12 @@ def _custom_init_fn(fieldsarg, frozen, has_post_init, self_name, globals):
     )
 
 
+# register all ORSO classes here:
+ORSO_DATACLASSES = dict()
+
+
 def orsodataclass(cls: type):
+    ORSO_DATACLASSES[cls.__name__] = cls
```

Comment:
I think it would be cleaner to use in-build python functionality instead of keeping track of our sub-classes ourselfs. All header classes are derived from `Header` so you could replace:

```python
if value.__class__ in ORSO_DATACLASSES.values():
```
with
```python
if isinstance(value, Header):
```

and

```python
cls = ORSO_DATACLASSES[ORSO_class]
```
with
```python
orso_classes = dict((c.__name__, c) for c in Header.__subclasses__()))
...
  cls = orso_classes[ORSO_class]
```
(I think the first call misses sub-subclasses, so it's probably easier to implement it as recursive method in the Header class.)

In this case we are also safe with people sub-classing orso Header without using a decorator (e.g. adding functionality not attributes).

Reply:
The subclass recursive search would miss OrsoDataset, which needs to be recognized as an ORSO class during deserialization.  I think trying to make an inheritance tree that includes both Header and OrsoDataset is going to be more complicated.